apps/molgenis-components/src/components/utils.ts

```diff
@@ -39,6 +40,60 @@ export function flattenObject(object: Record<string, any>): string {
   }
 }
 
+export async function getPrimaryKeys(
+  rows: IRow[],
+  tableName: string,
+  schemaName: string
+): Promise<(IRow | null)[]> {
+  return await Promise.all(
+    rows.map(async (row: IRow) => {
+      return await requestPrimaryKey(row, tableName, schemaName);
+    })
+  );
+}
+
+export async function requestPrimaryKey(
+  row: IRow,
+  tableName: string,
+  schemaName: string
+): Promise<IRow | null> {
+  const client = Client.newClient(schemaName);
```

Comment:
why create a new client for each key request ? , the idea of the client is to have a stateful object that only fetches the schema data once. Suggest to move this function into the client, or pass the schema-data ( instead of the schema name ) then the function can be synchronous.